File: backend/EventosAPI.Domain/Enums/Enums.cs
namespace EventosAPI.Domain.Enums
{
    public enum UserRole
    {
        User,
        Admin,
        Gatekeeper
    }

    public enum PaymentMethod
    {
        Pix,
        Cash,
        Card,
        Other
    }

    public enum OrderStatus
    {
        Pending,
        Paid,
        Canceled,
        Expired
    }

    public enum TicketType
    {
        Student,
        Staff,
        Other
    }

    public enum TicketStatus
    {
        Pending,
        Paid,
        Used,
        Expired,
        Canceled
    }

    public enum BatchType
    {
        Quantity,
        Time,
        Free
    }

    public enum EventCategory
    {
        Technology,
        Business,
        Education,
        Entertainment,
        Sports,
        Health,
        Other
    }
}

File: Frontend/src/types/enums.ts
export enum EventCategory {
  Technology = 'Technology',
  Music = 'Music',
  Sports = 'Sports',
  Arts = 'Arts',
  Food = 'Food',
  Business = 'Business',
  Education = 'Education',
  Entertainment = 'Entertainment',
  Other = 'Other'
}

export enum BatchType {
  Free = 'Free',
  TimeWindow = 'TimeWindow',
  Quantity = 'Quantity'
}

export enum EventRoleType {
  None = 0,
  ManageEvent = 1,
  ManageTickets = 2,
  ManageRoles = 4,
  ManageBatches = 8,
  ViewReports = 16,
  Moderator = 32 // All permissions
}

export enum UserRole {
  USER = 1,
  ADMIN = 2,
  GATEKEEPER = 3,
}

export enum PaymentMethod {
  Free = 'Free',
  CreditCard = 'CreditCard',
  DebitCard = 'DebitCard',
  PIX = 'PIX',
  BankTransfer = 'BankTransfer'
}

export enum OrderStatus {
  Pending = 'Pending',
  Reserved = 'Reserved',
  Paid = 'Paid',
  Canceled = 'Canceled',
  Expired = 'Expired'
}

export enum TicketStatus {
  PENDING = 'Pending',
  PAID = 'Paid',
  USED = 'Used',
  CANCELED = 'Canceled',
  EXPIRED = 'Expired'
}

export enum TicketType {
  Regular = 'Regular',
  VIP = 'VIP',
  Student = 'Student',
  Other = 'Other'
}
